pkg/monitoring/metrics/virt-controller/vmstats_collector.go

```diff
@@ -587,3 +597,78 @@ func collectVMCreationTimestamp(vms []*k6tv1.VirtualMachine) []operatormetrics.C
 
 	return cr
 }
+
+func CollectVmsVnicInfo(vms []*k6tv1.VirtualMachine) []operatormetrics.CollectorResult {
+	var results []operatormetrics.CollectorResult
+
+	for _, vm := range vms {
+		if vm.Spec.Template == nil || vm.Spec.Template.Spec.Domain.Devices.Interfaces == nil {
+			continue
+		}
+
+		interfaces := vm.Spec.Template.Spec.Domain.Devices.Interfaces
+		networks := vm.Spec.Template.Spec.Networks
+
+		for _, iface := range interfaces {
+			bindingType, bindingName := getBinding(iface)
+			networkName, matchFound := getNetworkName(iface.Name, networks)
+
+			if !matchFound {
+				continue
+			}
+
+			results = append(results, operatormetrics.CollectorResult{
+				Metric: vmVnicInfo,
+				Labels: []string{
+					vm.Name,
+					vm.Namespace,
+					iface.Name,
+					bindingType,
+					networkName,
+					bindingName,
+				},
+				Value: 1.0,
+			})
+		}
+	}
+
+	return results
+}
+
+func getBinding(iface k6tv1.Interface) (bindingType, bindingName string) {
+	bindingType = none
+	bindingName = none
```

Comment:
@avlitman I think you can simplify this with
const none = "none"
for both functions and avoid duplication

Reply:
we agreed to change to "" and its Go default to a string, so removed the initial values for both functions.